ui/src/connection.ts

```diff
@@ -136,26 +137,50 @@ export function generateSecret(): string {
   return randomID(SECRET_LEN)
 }
 
-async function rawCommand<T>(baseUrl: string, command: unknown): Promise<T> {
-  return fetch(`${baseUrl}/command`, {
-    method: 'POST',
-    mode: 'same-origin',
-    headers: {
-      'Content-Type': 'application/json',
-    },
-    body: JSON.stringify(command),
-  }).then(res => {
-    if (res.status !== 200) {
-      throw new Error('Unexpected status code: ' + res.status);
+async function rawCommand<T>(baseUrl: string, command: unknown, attempt?: number): Promise<T> {
+  if (attempt === undefined) attempt = 1
+
+  const retry = async () => {
+    const wait = Math.min((2 ** (attempt! - 1)) * 100, 10000)
+    await sleep(wait)
+    return rawCommand<T>(baseUrl, command, attempt! + 1)
+  }
+
+  let res;
+  try {
+    res = await fetch(`${baseUrl}/command`, {
+      method: 'POST',
+      mode: 'same-origin',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify(command),
+    })
+  } catch(e) {
+    // console.error('API command error: ', e);
+    // console.log(e)
+    // throw e;
+    return await retry()
+  }
+
+  if (res.status !== 200) {
+    const body = await res.text()
+
+    if (body === DROPPED_SESSION_BODY) {
+      // TODO: Do Hello() again (prob not smart because the whole room has been deleted)
+      //  or trigger a new onError event instead.
+      throw new Error('Session has expired.')
     }
 
-    return res.json()
-  }).catch(e => {
-    console.error('API command error: ', e);
-    throw e;
-  });
-}
+    // TODO: Only retry known networking errors?
+    return await retry()
```

Comment:
networking errors are caught at line 163 already, here you need to check if this is a 5xx error. There's indeed to point in retrying 4xx or any other kind of error :)

Reply:
yup 👍